pennylane/templates/subroutines/single_excitation_op.py

```diff
@@ -0,0 +1,160 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the ``SingleExcitationOp`` template.
+"""
+# pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
+from pennylane.ops import RX, RZ, Hadamard, CNOT 
+from pennylane.templates.decorator import template
+from pennylane.templates.utils import (
+    check_no_variable,
+    check_shape,
+    check_wires,
+    check_type,
+    get_shape,
+)
+
+from pennylane import numpy as np
+
+@template
+def SingleExcitationOp(weight, wires=None):
```

Comment:
This name doesn't really put in my mind that this is actually a unitary/circuit. I suggest seeing if we can come up with a clearer name.

Some possibilities:
`SingleExcitationUnitary`
`SingleExcitationTransformation`
`SingleExcitationEvolution`
or similar

Reply:
``SingleExcitationUnitary`` is good. Thanks @co9olguy .